src/avro/decode.rs

```diff
@@ -244,6 +301,75 @@ pub fn decode<'a, R: AsyncRead + Unpin + Send>(
                     Err(DecodeError::new("enum symbol not found").into())
                 }
             }
+            SchemaPiece::Fixed { size, .. } => {
+                let mut buf = vec![0u8; *size as usize];
+                reader.read_exact(&mut buf).await?;
+                Ok(Value::Fixed(*size, buf))
+            }
+            SchemaPiece::ResolveRecord {
+                defaults,
+                fields,
+                n_reader_fields,
+            } => {
+                let mut items: Vec<Option<(String, Value)>> = vec![None; *n_reader_fields];
+                for default in defaults {
+                    if items[default.position].is_some() {
```

Comment:
this should be `items.get(default.position).is_some()`, current code will panic (I guess if it's a vec and not some custom Index type).

hmm actually I see that you are setting this to an option, not sure how possible it is for the `items` list to be too short? Might be fine and intentional, just looking like a bug?

Reply:
This is intentional; I will add an assert to make it more clear.